client.go

```diff
@@ -25,9 +25,62 @@ type Client struct {
 	BaseURL      string
 	StreamingURL string
 	hc           *http.Client
+	token        *oauth2.Token
+}
+
+// WithToken provides an oauth2.Token to the client for auth.
+func WithToken(t *oauth2.Token) func(c *Client) error {
+	return func(c *Client) error {
+		c.token = t
+		return nil
+	}
+}
+
+// WithTokenFile reads a JSON serialized oauth2.Token struct from disk and provides it
+// to the client for auth.
+func WithTokenFile(path string) func(c *Client) error {
+	t, err := loadToken(path)
+	if err != nil {
+		return func(c *Client) error {
+			return err
+		}
+	}
+	return WithToken(t)
+}
+
+// New creates a new Tesla API client. You must provided one of WithToken or WithTokenFile
+// functional options to initialize the client with an OAuth token.
+func New(ctx context.Context, options ...func(*Client) error) (*Client, error) {
+	config := &oauth2.Config{
```

Comment:
outside the scope of this change, but I'm expecting at some point in the future we'll want a WithOAuth2Config option that allows setting a different config.

Reply:
https://github.com/bogosj/tesla/issues/31